reco_utils/recommender/ripplenet/model.py

```diff
@@ -0,0 +1,241 @@
+# This code is modified from RippleNet
+# Online code of RippleNet: https://github.com/hwwang55/RippleNet
+
+import tensorflow as tf
+import numpy as np
+from sklearn.metrics import roc_auc_score
+
+class RippleNet(object):
```

Comment:
please take a look at the class that Jianjun did https://github.com/microsoft/recommenders/blob/master/reco_utils/recommender/deeprec/models/base_model.py#L17. 

I think we can add a couple of things: one is the seed, setting up the numpy and tf seed. Also this is important:
```
        # set GPU use with demand growth
        gpu_options = tf.GPUOptions(allow_growth=True)
        self.sess = tf.Session(config=tf.ConfigProto(gpu_options=gpu_options))
        self.sess.run(self.init_op)
```
this doesn't lock the gpu and also sets internally the session, so now we don't have to do: 
```
with session:
   model.fit()
```
just `model.fit`